tools/hardklor/hardklor.xml

```diff
@@ -0,0 +1,75 @@
+<tool id="hardklor" name="Hardklor" version="2.30">
+    <description>identification of features from mass spectra</description>
+    <requirements>
+        <requirement type="package" version="2.30">hardklor</requirement>
+    </requirements>
+    <stdio>
+        <exit_code range="1:" level="fatal"/>
+    </stdio>
+    <command>python hardklor.py $ms1_in.extension $ms1_in
+    $output -charge_algo=$cdm -resolution=$res -centroided=$centroided -instrument=$instrument -algorithm=$algorithm -mincharge=$chMin 
+    -maxcharge=$chMax -mslvl=$mslvl -correlation=$corr -depth=$depth -sensitivity=$sens -maxfeat=$maxfeat</command>
+    <inputs>
+	<param name="ms1_in" type="data" format="mzml" label="MS1 files with precursor spectra " />
+        <param name="res" type="integer" value="60000" label="Spectral resolution at 400m/z" />
+        <param name="centroided" type="select" display="radio" label="Centroided data?">
+          <option value="0" selected="true">No</option>
+          <option value="1">Yes</option>
+        </param>
+        <param name="instrument" type="select" label="Instrument" >
+            <option value="Orbitrap" selected="true">Orbitrap</option>
+            <option value="FTICR" >FTICR</option>
+            <option value="TOF" >TOF</option>
+            <option value="QIT" >Quadrupole ion trap</option>
+        </param>
+        <param name="mslvl" type="select" display="radio" label="MS level">
+          <option value="0" >All</option>
+          <option value="1" selected="true">MS1</option>
+          <option value="2">MS2</option>
+          <option value="3">MS3</option>
+        </param>
+        <param name="algorithm" type="select" label="Algorithm">
+            <option value="Basic">Basic</option>
+            <option value="Version1">Version 1</option>
+            <option value="Version2" selected="true">Version 2</option>
+        </param>
+        <param name="cdm" type="select" label="Charge determination method">
+            <option value="None">None</option>
```

Comment:
Apparently Hardklor accepts a value "None" for the charge determination in its config file.